Amend Load Validations - V3

luke-hill opened the pull request:
Remove default load validation and configuration options.

I've opted not to go down the deprecation route, as this is something I only added in as a configuration option to fix the issue of URL-less pages.

luke-hill commented:
ping @tgaff @RustyNail <- If you're interested in reviewing stuff for v3

ineverov commented on spec/loadable_spec.rb:
```diff
@@ -41,33 +41,11 @@
-    end
+    context 'A standard Page' do
+      it 'has no default load validations' do
+        klass = Class.new(SitePrism::Page)
```
hah! You said you prefer other way of doing this some time [ago](https://github.com/natritmeyer/site_prism/pull/250#discussion_r186646007) 😆 
North remembers!

luke-hill replied:
This area I'm not entirely sure how to get it working the way I want. If you can point me in the right way I'll fix it up.

It "may" be that I can actually fix this now there is only one code path not 2 (0 / 1 default validations). For some of the Loadable tests here, they create phantom classes that just inherit the Loadable module (And are several years old), so I'm loathe to change them.

I'll have a look next week to see if I can standardise this.

luke-hill commented:
ping @ineverov / @tgaff - I have ...
- Rebased this against the latest v3 alpha work.
- Fixed up documentation / CR comments
- Tidied up some linelength stuff in readme

Let me know what you think. I'll leave it in PR for a couple of weeks so you've got time to look!

ineverov commented on README.md:
```diff
@@ -1465,14 +1465,13 @@ end
-NOTE: `SitePrism::Page` includes a default load validation on `page.displayed?` which is applied
-to all pages.  It is therefore not necessary to define a load validation for this condition on
-inheriting page objects.
+**NB:** `SitePrism::Page` **used to** include a default load validation on `page.displayed?` however for
```
👍
